### PR TITLE
Add gsa locations reusing locations table

### PR DESCRIPTION
**Pull request by scott062**

### CODE REVIEW

**Comment by scott062** [comment]
Just pushed final changes @elizabeth-jimenez

File: src/services/lookups.js [hunk 1]
```diff
@@ -55,6 +55,51 @@ const getLocations = Locations => async () => {
...
 
+const mapGSALocQuery = q => {
+  const columns = {
+    'location_code': q?.['rp.locgvtgeoloccd'],
+    'location_state': q?.['rp.locstate'],
+    'location_state': q?.['rp.locgvtstcntrydescr'],
+    'location_city': q?.['rp.loccity'],
+    'location_country': q?.['rp.loccountry'],
+  }
+  for (let key in columns) {
```
Comment (elizabeth-jimenez) [inline src/services/lookups.js:66]: Gross! ;)

File: src/services/lookups.js [hunk 2]
```diff
@@ -55,6 +55,51 @@ const getLocations = Locations => async () => {
...
+    const results = data.serialize().map(d => {
+      return {
+        "locgvtgeoloccd": d.location_code,
+        "loceffdt": "",
+        "loceffstatus": "A",
+        "locgvtstcntrydescr": d.location_state,
+        "loccity": d.location_city,
+        "locstate": d.location_state?.slice(0, 2).toUpperCase(),
+        "loccounty": " ",
+        "loccountry": d.location_country,
```
Comment (elizabeth-jimenez) [inline src/services/lookups.js:90]: loccity and loccountry are also caps, but i only confirmed on 2 - will dbl check.

File: src/services/common.js [hunk 3]
```diff
@@ -465,8 +479,8 @@ const convertTemplateFiltersCols = (query, mapFunc) => {
     const f$ = f.split('|');
 
     let name = f$[0].toLowerCase()
-    if (mapFunc){
-      name = mapFunc([f$[0].toLowerCase()])[0]
+    if (mapFunc) {
+      name = mapFunc([name])?.[0]
```
Comment (elizabeth-jimenez) [inline src/services/common.js:483]: 🙌

File: src/services/lookups.js [hunk 4]
```diff
@@ -55,6 +56,42 @@ const getLocations = Locations => async () => {
...
 }
 
+const getGSALocations = async (query) => {
+  try {
+    const filsCols = common.convertTemplateFiltersCols(query, x => x.map(common.gsaNameMapping))
+
+    let query$ = filsCols?.filters.reduce((result, currentFilter) => {
+      result[currentFilter?.name] = currentFilter?.value;
+      return result;
+    }, {});
```
Comment (elizabeth-jimenez) [inline src/services/lookups.js:66]: wow - so this whole time i've been assuming `.where` needs a '=', but seems it is the default 😭

File: src/services/lookups.js (hunk 2 above)
Comment (elizabeth-jimenez) [reply src/services/lookups.js:88]: confirmed - and the country is the abbreviation of it, but this is mock, so do as ya plz 🔥

File: src/services/common.js [hunk 5]
```diff
@@ -377,6 +377,20 @@ const asgNameMapping = (val, toWS=false) => {
...
+    locgvtgeoloccd: 'location_code',
+    locstate: 'location_state',
+    locgvtstcntrydescr: 'location_state',
+    loccity: 'location_city',
+    loccountry: 'location_country'
+  };
+  if(toWS) {
+    colDictionary = _.invert(colDictionary);
+  }
+  return colDictionary?.[val] || val
```
Comment (elizabeth-jimenez) [inline src/services/common.js:391]: Reducing reliance on Lodash, very nice

File: src/services/lookups.js [hunk 6]
```diff
@@ -55,6 +56,48 @@ const getLocations = Locations => async () => {
   }
 }
 
+const getGSALocations = async (query) => {
+  try {
+    if (query['rp.columns'] === 'ROWCOUNT') {
```
Comment (elizabeth-jimenez) [inline src/services/lookups.js:61]: Nice addition - reduces gaps in mock functionality

**Review by elizabeth-jimenez — APPROVED** [review]
Very not reducing 🤌

**Pull request merged**